packages/editor-ui/src/mixins/expressionManager.ts

```diff
@@ -71,6 +71,8 @@ export const expressionManager = defineComponent({
 		},
 
 		segments(): Segment[] {
+			if (!this.editor || !this.editor.state) return [];
```

Comment:
1. We can simplify the check with `?`
2. Why is this check needed now and not before?

Reply:
We added a preview component which uses `segments` [in the template](https://github.com/n8n-io/n8n/blob/ba98ef15eb19474964b0e4cabe7ed64f1041019f/packages/editor-ui/src/components/SqlEditor/SqlEditor.vue#L5). This causes `null` to be sent as `this.editor` sometimes. So we can either add this check here or when passing `segments` prop to the `InlineExpressionEditorOutput`. I chose this option so we are not adding unnecessary logic to vue templates.

Reply:
Thanks! Point 1 is still pending.